### ISSUE TITLE
Aggregate constructor treats object as empty pipeline.

### ISSUE THREAD

**Issue by aksel** [Automattic/mongoose#6732]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
The Aggregate constructor used to accept spreads. (#2716). This meant that the following was a valid pipeline.

    SomeModel
        .aggregate({ $match: { someField: 'xxxx' })
        .then(docs => console.log(docs); // Documents matching criteria
            

Since 5.0.0-rc0, this was removed, and was reported as a breaking change. If you pass an object as the first parameter, it treats it as an empty pipeline. Outputting after an empty pipeline throws an error (Error: Aggregate has empty pipeline).

However, if you chain another pipeline stage, the error will be swallowed, and the aggregation will go on as if the constructor was empty.

**If the current behavior is a bug, please provide the steps to reproduce.**
The following would return all documents in a collection, if it has the memory to sort it.

    SomeModel
        .aggregate({ $match: { someField: 'xxxx' })
        .sort({ timestamp: -1 })
        .then(docs => console.log(docs); // All documents in collection!

**What is the expected behavior?**
I think passing an invalid argument to the constructor should throw an error, instead of treating it an an empty constructor.

I am honestly unsure whether you can consider this a bug or not.

**Please mention your node.js, mongoose and MongoDB version.**
node: 10.6.0

mongoose: 5.2.3

MongoDB: 3.6.5

**Comment by vkarpov15** [comment]
You're right, this is an easy detail to mix up. We'll fix and throw a helpful error in this case.
